src/pkg/boltz/boltzService.ts

```diff
@@ -0,0 +1,372 @@
+import type { CreateSwapCommonResponse, NetworkString, SubmarineSwapResponse } from './boltz';
+import { Boltz } from './boltz';
+import { randomBytes } from 'crypto';
+import type { BIP174SigningData, OwnedInput } from 'liquidjs-lib';
+import {
+  address,
+  AssetHash,
+  Blinder,
+  Creator,
+  CreatorInput,
+  crypto,
+  Extractor,
+  Finalizer,
+  networks,
+  Pset,
+  script as bscript,
+  script,
+  Signer,
+  Transaction,
+  Updater,
+  witnessStackToScriptWitness,
+  ZKPGenerator,
+  ZKPValidator,
+} from 'liquidjs-lib';
+import type { TagData } from 'bolt11';
+import bolt11 from 'bolt11';
+import type { Unspent } from '../../domain/chainsource';
+import type { ECPairInterface } from 'ecpair';
+import zkp from '@vulpemventures/secp256k1-zkp';
```

Comment:
Ideally, you should let the user pass the `ZKPInterface` in constructor in order to inject its own zkp lib implementation. In other words `await zkp()` should be done outside of that module